tools/scoring/score/dimensions/dimension.py

```diff
@@ -176,8 +180,123 @@ def is_entity_virtual(*, entity: EntityInstance) -> bool:
     """
     return entity.links is not None
 
+  @staticmethod
+  def match_virtual_entities(
+      *, solution_points_virtual: List[Tuple[Set[str], EntityType]],
+      proposed_points_virtual: List[Tuple[Set[str], EntityType]],
+      sort_candidates_by_key: str) -> Dict[float, List[Dict]]:
+    """
+    Finds the closest correlating virtual entities between two files.
```

Comment:
can you please add details how it is doing that?
tracking through the cloud device id .....